src/Platform/Microsoft.Testing.Platform/Hosts/TestHostBuilder.cs

```diff
@@ -742,36 +742,14 @@ private static async Task<ITestFramework> BuildTestFrameworkAsync(TestFrameworkB
 
         IDataConsumer[] dataConsumerServices = dataConsumersBuilder.ToArray();
 
-        // Build the message bus
-        // If we're running discovery command line we don't want to process any messages coming from the adapter so we filter out all extensions
-        // adding a custom message bus that will simply forward to the output display for better performance
-        if (serviceProvider.GetCommandLineOptions().IsOptionSet(PlatformCommandLineProvider.DiscoverTestsOptionKey)
-            // In case of server mode the discovery is handled by the ServerHost using the standard message bus
-            && !testFrameworkBuilderData.IsJsonRpcProtocol)
-        {
-            ListTestsMessageBus concreteMessageBusService = new(
-                serviceProvider.GetTestApplicationCancellationTokenSource(),
-                serviceProvider.GetLoggerFactory(),
-                serviceProvider.GetOutputDevice(),
-                serviceProvider.GetAsyncMonitorFactory(),
-                serviceProvider.GetEnvironment(),
-                serviceProvider.GetTestApplicationProcessExitCode(),
-                pushOnlyProtocol,
-                pushOnlyProtocolDataConsumer);
-            await concreteMessageBusService.InitAsync();
-            testFrameworkBuilderData.MessageBusProxy.SetBuiltMessageBus(concreteMessageBusService);
-        }
-        else
-        {
-            AsynchronousMessageBus concreteMessageBusService = new(
-                dataConsumerServices,
-                serviceProvider.GetTestApplicationCancellationTokenSource(),
-                serviceProvider.GetTask(),
-                serviceProvider.GetLoggerFactory(),
-                serviceProvider.GetEnvironment());
-            await concreteMessageBusService.InitAsync();
-            testFrameworkBuilderData.MessageBusProxy.SetBuiltMessageBus(concreteMessageBusService);
-        }
+        AsynchronousMessageBus concreteMessageBusService = new(
```

Comment:
this removes the first leg of the if. it is possible to implement all this with the custom message bus, but it means duplicating the logic that output device already has. And this makes it similar to how hotreload works.

Reply:
I am not able to understand what's the impact of the change so I'll trust you and say that if that's green it's good for me.

Reply:
The comment was saying:

> adding a custom message bus that will simply forward to the output display for better performance

Not sure if it's still relevant?